### PR TITLE
[DOC] Update agent API

### PR DESCRIPTION
**Pull request by pingsutw**
Update agent API and example in the Doc

https://flyte--4943.org.readthedocs.build/en/4943/flyte_agents/testing_agents_locally.html

### CODE REVIEW

File: docs/flyte_agents/developing_agents.md [hunk 1]
```diff
@@ -25,61 +25,70 @@ While agents can be written in any programming language, we currently only suppo
...
 
-class CustomAsyncAgent(AsyncAgentBase):
-    def __init__(self, task_type: str):
-        # Each agent should have a unique task type.
-        # The Flyte agent service will use the task type
-        # to find the corresponding agent.
-        self._task_type = task_type
+class BigQueryAgent(AsyncAgentBase):
+    def __init__(self):
+        super().__init__(task_type_name="saprk", metadata_type=BigQueryMetadata)
```
Comment (Future-Outlier) [inline docs/flyte_agents/developing_agents.md:48]:
```suggestion
        super().__init__(task_type_name="bigquery", metadata_type=BigQueryMetadata)
```

File: docs/flyte_agents/developing_agents.md [hunk 2]
```diff
@@ -25,61 +25,70 @@ While agents can be written in any programming language, we currently only suppo
...
 # To register the custom agent
-AgentRegistry.register(CustomAsyncAgent())
+AgentRegistry.register(BigQueryAgent())
 ```
 
 For an example implementation, see the [BigQuery agent](https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-bigquery/flytekitplugins/bigquery/agent.py#L43).
+
+## Sync agent interface specification
+
+To create a new async agent, extend the `SyncAgentBase` class and implement `do` methods. This method need to be idempotent.
```
Comment (Future-Outlier) [inline docs/flyte_agents/developing_agents.md:75]:
```suggestion
To create a new sync agent, extend the `SyncAgentBase` class and implement `do` methods. This method needs to be idempotent.
```

File: docs/flyte_agents/developing_agents.md [hunk 3]
```diff
@@ -25,61 +25,70 @@ While agents can be written in any programming language, we currently only suppo
 
 ## Async agent interface specification
 
-To create a new async agent, extend the `AgentBase` class in the `flytekit.backend` module and implement `create`, `get`, and `delete` methods. All calls must be idempotent.
+To create a new async agent, extend the `AsyncAgentBase` and implement `create`, `get`, and `delete` methods. These methods need to be idempotent.
```
Comment (neverett) [inline docs/flyte_agents/developing_agents.md:28]:
```suggestion
To create a new async agent, extend the `AsyncAgentBase` and implement `create`, `get`, and `delete` methods. These methods must be idempotent.
```

File: docs/flyte_agents/developing_agents.md (hunk 2 above)
Comment (neverett) [inline docs/flyte_agents/developing_agents.md:75]:
```suggestion
To create a new sync agent, extend the `SyncAgentBase` class and implement a `do` method. This method must be idempotent.
```

File: docs/flyte_agents/testing_agents_locally.md [hunk 4]
```diff
@@ -24,14 +24,15 @@ For example, you need to set the `GOOGLE_APPLICATION_CREDENTIALS` environment va
 
 ```
 
+Adding `AsyncAgentExecutorMixin` to this class to tell flytekit to use the agent to run the task.
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_locally.md:27]:
```suggestion
Add `AsyncAgentExecutorMixin` to this class to tell flytekit to use the agent to run the task.
```

File: docs/flyte_agents/testing_agents_locally.md [hunk 5]
```diff
@@ -44,5 +45,36 @@ bigquery_doge_coin = BigQueryTask(
...
+    sess = flytekit.current_context().spark_session
+    count = (
+        sess.sparkContext.parallelize(range(1, n + 1), partitions).map(f).reduce(add)
+    )
+    pi_val = 4.0 * count / n
+    print("Pi val is :{}".format(pi_val))
+    return pi_val
+```
+
+To execute the Spark task on the agent, it's essential to configure the raw-output-data-prefix with a remote path.
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_locally.md:69]:
```suggestion
To execute the Spark task on the agent, you must configure the `raw-output-data-prefix` with a remote path.
```

File: docs/flyte_agents/testing_agents_locally.md [hunk 6]
```diff
@@ -44,5 +45,36 @@ bigquery_doge_coin = BigQueryTask(
...
+This configuration ensures that Flytekit transfers the input data to the blob storage.
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_locally.md:70]:
```suggestion
This configuration ensures that flytekit transfers the input data to the blob storage and allows the Spark job running on Databricks to access the input data directly from the designated bucket.
```

File: docs/flyte_agents/testing_agents_locally.md [hunk 7]
```diff
@@ -44,5 +45,36 @@ bigquery_doge_coin = BigQueryTask(
...
+Consequently, the Spark job running on Databricks can access the input data directly from the designated bucket.
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_locally.md:71]:
```suggestion

```

File: docs/flyte_agents/testing_agents_locally.md [hunk 8]
```diff
@@ -44,5 +45,36 @@ bigquery_doge_coin = BigQueryTask(
...
+
+```{note}
+The Spark task will run locally if the raw-output-data-prefix is not set.
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_locally.md:74]:
```suggestion
The Spark task will run locally if the `raw-output-data-prefix` is not set.
```

**Review by neverett — CHANGES_REQUESTED** [review]
Some small edits requested, otherwise looks great, thanks for adding this!

**Review by neverett — APPROVED** [review]

**Pull request merged**